sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -456,11 +456,25 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
     val keyExpr = df.col(col.name).expr
     def buildExpr(v: Any) = Cast(Literal(v), keyExpr.dataType)
     val branches = replacementMap.flatMap { case (source, target) =>
-      Seq(buildExpr(source), buildExpr(target))
+        if (!(checkIfNonFractionalNumeric(col.dataType) &&
+          (isNaN(source) || isNaN(target)))) {
+          Seq(buildExpr(source), buildExpr(target))
```

Comment:
`Seq` or `None` seems a bit weird - shouldn't it be `Seq(stuff) else Seq.empty`?

Reply:
Mostly because `Seq` and `None` are two different data types, one is a `Seq` and the other is implicitly an `Option`.